include/psa/crypto_extra.h

```diff
@@ -1344,6 +1344,70 @@ psa_status_t psa_crypto_driver_pake_get_role(
     const psa_crypto_driver_pake_inputs_t *inputs,
     psa_pake_role_t *role);
 
+/** Get the lengths of the user id in bytes from given inputs.
```

Comment:
I may not remember all the history of discussions here but in #5451 which has been a reference to define the PAKE driver interface, there are two ways for the driver to get a buffer of early data.
Through two APIs:
```C
psa_status_t psa_crypto_driver_key_derivation_get_input_size(
    const psa_crypto_driver_key_derivation_inputs_t *inputs,
    psa_key_derivation_step_t step,
    size_t *size);
psa_status_t psa_crypto_driver_key_derivation_get_input_bytes(
    const psa_crypto_driver_key_derivation_inputs_t *inputs,
    psa_key_derivation_step_t step,
    uint8_t *buffer, size_t buffer_size, size_t *buffer_length);
```
Through only one:
```C
psa_status_t psa_crypto_driver_key_derivation_get_input_key(
    const psa_crypto_driver_key_derivation_inputs_t *inputs,
    psa_key_derivation_step_t step,
    uint8_t** p_key_buffer, size_t *key_buffer_size);
```
With the second one you give a chance to the driver to avoid a memory allocation. And as far as I remember the first way was introduced in the case we need to generate on-the-fly data in the core (but again I may not remember well or all the reasons).
Here I would say that the second way (only one function) is better suited. I know that for the password we've gone the first way but I am not sure now it was the right call (no need to change it in this PR though).

Reply:
> We are using the first way as it is designed for for transparent drivers and the second form is for opaque drivers

Why do you say so?

Reply:
Please see https://github.com/Mbed-TLS/mbedtls/pull/6567#discussion_r1082953367.

Reply:
I am not convinced yet that we can infer that the first way is designed for transparent drivers and the second for opaque drivers. Anyway, better for the get password/peer/user to be aligned thus let's keep it that and I am creating an issue to decide if we keep things as they are or if we move to the second way.